react/features/whiteboard/middleware.ts

```diff
@@ -0,0 +1,222 @@
+        const conference = getCurrentConference(state);
+        const isEnabled = isWhiteboardOpen(state);
+        let collabLink = null;
+
+        if (!isEnabled) {
+            collabLink = getCollabLink(state);
+
+            if (!Object.keys(collabLink).length) {
+                collabLink = await generateCollaborationLinkData();
+            }
+        }
+
+        dispatch(setTileView(false));
+        sendWhiteboardCommand({
+            ...collabLink ? {
+                roomId: getCurrentRoomId(state),
+                roomKey: collabLink?.roomKey
+            } : {},
+            conference,
+            participantId: localParticipantId,
+            whiteboardId: action.id,
+            isEnabled: !isEnabled
+        });
+
+        break;
+    }
+    case CONFERENCE_JOIN_IN_PROGRESS: {
+        const { conference } = action;
+
+        // @ts-ignore
+        conference.addCommandListener(WHITEBOARD, ({ value, attributes }) => {
+            handleWhiteboardStatus(store, conference, value, attributes);
+        });
+        break;
+    } // TODO: cleanup PARTICIPANT_DISPLAY_NAME_CHANGED
+    case PARTICIPANT_UPDATED: {
+        if (typeof APP !== 'undefined') {
+            const participant = getLocalParticipant(store.getState());
+
+            if (participant
+                && participant.id === action.participant.id
+                && action.participant.name
+                && participant.name !== action.participant.name) {
+                localStorage.setItem(
+                    WHITEBOARD_STORAGE_KEY,
+                    JSON.stringify({ username: action.participant.name })
+                );
+                dispatch(setUsernameStatus(USERNAME_UPDATE_STATUS.NEW));
+            }
+        }
+        break;
+    }
+    case PARTICIPANT_LEFT: {
+        const whiteboardId = getWhiteboardId(state);
+
+        if (action.participant.id === whiteboardId) {
+            dispatch(disableWhiteboard());
+        }
+        break;
+    }
+    }
+
+    return next(action);
+});
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, e.g. Close the whiteboard if it's left open.
+ */
+StateListenerRegistry.register(
+
+    // @ts-ignore
+    state => getCurrentConference(state),
+
+    // @ts-ignore
+    (conference, { dispatch }, previousConference): void => {
+        if (previousConference) {
+
+            dispatch(disableWhiteboard());
+        }
+    });
+
+
+/**
+ * Handles the whiteboard state.
+ *
+ * @param {Store} store - The redux store.
+ * @param {JitsiConference} conference - The current conference.
+ * @param {string} whiteboardId - The current whiteboard id.
+ * @param {Object} attributes - The command attributes.
+ * @returns {void}
+ */
+
+// @ts-ignore
+function handleWhiteboardStatus(store: IStore, conference, whiteboardId: string, { isEnabled, roomId, roomKey, from }) {
+    const { dispatch, getState } = store;
+
+    if (isEnabled === 'false') {
```

Comment:
I'm not sure we want this behavior. Once the whiteboard has been created if the original creator closes it others should be able to continue using it.